src/lib.rs

```diff
@@ -81,84 +81,153 @@ pub enum ParameterType {
-    #[structopt(name = "git_ref", short = "t", long = "git_ref")]
-    pub git_ref: Option<String>,
+    #[arg(name = "git_ref", short = 't', long = "git_ref")]
+    git_ref: Option<String>,
 
     /// Specify the name of your generated project (and so skip the prompt asking for it)
-    #[structopt(name = "name", short = "n", long = "name")]
-    pub project_name: Option<String>,
+    #[arg(name = "name", short = 'n', long = "name")]
+    project_name: Option<String>,
 
     /// Specifiy the target directory
-    #[structopt(name = "target_directory", short = "d", long = "target_directory")]
-    pub target_dir: Option<PathBuf>,
+    #[arg(name = "target_directory", short = 'd', long = "target_directory")]
+    target_dir: Option<PathBuf>,
 
     /// Override target directory if it exists
-    #[structopt(short = "f", long = "force")]
-    pub force: bool,
+    #[arg(short = 'f', long = "force")]
+    force: bool,
 
     /// Append files in the target directory, create directory with the project name if it doesn't already exist but doesn't overwrite existing file (use force for that kind of usage)
-    #[structopt(short = "a", long = "append")]
-    pub append: bool,
+    #[arg(short = 'a', long = "append")]
+    append: bool,
 
     /// Ignored, kept for backwards compatibility [DEPRECATED]
-    #[structopt(short = "p", long = "passphrase")]
-    pub passphrase_needed: bool,
+    #[arg(short = 'p', long = "passphrase")]
+    passphrase_needed: bool,
 
     /// Specify if your private SSH key is located in another location than $HOME/.ssh/id_rsa
-    #[structopt(short = "k", long = "private_key_path")]
-    pub private_key_path: Option<PathBuf>,
+    #[arg(short = 'k', long = "private_key_path")]
+    private_key_path: Option<PathBuf>,
 
     /// Supply parameters via the command line in <name>=<value> format
-    #[structopt(long = "param")]
-    pub parameters: Vec<String>,
+    #[arg(long = "param")]
+    parameters: Vec<String>,
 }
 
-#[buildstructor::buildstructor]
 impl Opts {
-    #[builder]
-    #[allow(clippy::too_many_arguments)]
-    #[allow(dead_code)]
-    pub fn new(
-        template_path: PathBuf,
-        repository_template_path: Option<PathBuf>,
-        git_ref: Option<String>,
-        project_name: Option<String>,
-        target_dir: Option<PathBuf>,
-        force: Option<bool>,
-        append: Option<bool>,
-        private_key_path: Option<PathBuf>,
-        parameters: Vec<String>,
-    ) -> Opts {
-        Self {
-            template_path,
-            repository_template_path,
-            git_ref,
-            project_name,
-            target_dir,
-            force: force.unwrap_or_default(),
-            append: append.unwrap_or_default(),
-            passphrase_needed: false,
-            private_key_path,
-            parameters,
-        }
+    /// Set the template path for the structure.
+    pub fn template_path(mut self, path: &str) -> Self {
```

Comment:
Let's take a PathBuf directly

Reply:
Let's do it a generic `Into<PathBuf>` - because this allows users to pass `&str` as these string literals are handy already.